### PR TITLE
[SMALLFIX] Changed the thriftGen from project enumeration to detecting projects …

### PR DESCRIPTION
**Pull request by yupeng9**
…that contain thrift files

### CODE REVIEW

File: bin/alluxio [hunk 1]
```diff
@@ -319,7 +319,7 @@ case ${COMMAND} in
   exit 0
 ;;
 "thriftGen")
-  declare -a thriftSrcPaths=("${BIN}/../core/common/src" "${BIN}/../keyvalue/common/src")
+  declare -a thriftSrcPaths=($(find "${BIN}/.." -regex  ".*\.thrift" | awk -F/thrift/ '{print $(1)}' | sort |uniq))
```
Comment (aaudiber) [inline bin/alluxio]: `sort | uniq`

Comment (yupeng9) [inline bin/alluxio]: done.

**Comment by yupeng9** [comment]
@aaudiber any more comments?

**Comment by uronce-cc** [comment]
nice! LGTM

**Pull request merged**